File: e2e/elixir/test/engine_test.exs
# E2e tests for category: engine
defmodule E2e.EngineTest do
  use ExUnit.Case, async: true

  describe "engine_batch_basic" do
    test "CrawlEngine with defaults batch scrapes like the free function" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'batch.completed_count' not available on result type
      # skipped: field 'batch.total_count' not available on result type
    end
  end

  describe "engine_crawl_basic" do
    test "CrawlEngine with defaults crawls multiple pages like the free function" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'crawl.pages_crawled' not available on result type
      # skipped: field 'crawl.min_pages' not available on result type
    end
  end

  describe "engine_map_basic" do
    test "CrawlEngine with defaults discovers URLs like the free function" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'map.min_urls' not available on result type
    end
  end

  describe "engine_scrape_basic" do
    test "CrawlEngine with defaults scrapes a page identically to the free function" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      assert String.trim(result.status_code) == 200
      assert String.trim(result.content_type) == "text/html"
      assert String.trim(result.metadata.title) == "Engine Test"
      assert String.contains?(result.metadata.description, "Testing the engine")
      assert length(result.links) >= 1
      # skipped: field 'headings.h1_text' not available on result type
    end
  end

  describe "engine_stream_basic" do
    test "CrawlEngine with defaults streams events like the free function" do
      engine = Kreuzcrawl.create_engine!(nil)
      result = Kreuzcrawl.scrape!(engine, "")
      # skipped: field 'stream.has_page_event' not available on result type
      # skipped: field 'stream.has_complete_event' not available on result type
      # skipped: field 'stream.event_count_min' not available on result type
    end
  end
end

File: e2e/node/tests/engine.test.ts
import { describe, it, expect } from 'vitest';
import { scrape, createEngine } from '@kreuzberg/kreuzcrawl';

describe('engine', () => {
  it('engine_batch_basic: CrawlEngine with defaults batch scrapes like the free function', async () => {
    const engine = createEngine(null);
    const result = await scrape(engine, "");
    // skipped: field 'batch.completed_count' not available on result type
    // skipped: field 'batch.total_count' not available on result type
  });

  it('engine_crawl_basic: CrawlEngine with defaults crawls multiple pages like the free function', async () => {
    const engine = createEngine(null);
    const result = await scrape(engine, "");
    // skipped: field 'crawl.pages_crawled' not available on result type
    // skipped: field 'crawl.min_pages' not available on result type
  });

  it('engine_map_basic: CrawlEngine with defaults discovers URLs like the free function', async () => {
    const engine = createEngine(null);
    const result = await scrape(engine, "");
    // skipped: field 'map.min_urls' not available on result type
  });

  it('engine_scrape_basic: CrawlEngine with defaults scrapes a page identically to the free function', async () => {
    const engine = createEngine(null);
    const result = await scrape(engine, "");
    expect(result.statusCode).toBe(200);
    expect(result.contentType).toBe("text/html");
    expect(result.metadata.title).toBe("Engine Test");
    expect(result.metadata.description).toContain("Testing the engine");
    expect(result.links.length).toBeGreaterThanOrEqual(1);
    // skipped: field 'headings.h1_text' not available on result type
  });

  it('engine_stream_basic: CrawlEngine with defaults streams events like the free function', async () => {
    const engine = createEngine(null);
    const result = await scrape(engine, "");
    // skipped: field 'stream.has_page_event' not available on result type
    // skipped: field 'stream.has_complete_event' not available on result type
    // skipped: field 'stream.event_count_min' not available on result type
  });
});
